### PR TITLE
feat: Add strict setting to opt-into v1 behavior

### LINKED ISSUE #137: No warning if only some imports from a module can be moved into a type-checking block
Let's start with the following example:

```
from collections.abc import Sequence
from collections.abc import Iterable

def f(a) -> Iterable:
    return []

def g(a) -> Sequence:
    return []
```

As expected, this results in the following warnings:
```
TC003 Move built-in import 'collections.abc.Sequence' into a type-checking block
TC003 Move built-in import 'collections.abc.Iterable' into a type-checking block
```

However, if we extend the example to:

```
from collections.abc import Sequence
from collections.abc import Iterable

def f(a) -> Iterable:
    if isinstance(a, Sequence):
        ...
    return []

def g(a) -> Sequence:
    return []
```

There are no warnings anymore even though one of the imports can be moved into a type-checking block.

In python-poetry/poetry#6807, you can see that such a warning could have even prevented a real bug. There is an incompatibility between poetry 1.2.2 and packaging 20.4 because of an import of `NormalizedName` from `packaging.utils`, which could have been moved into a type-checking block. However, there is no warning because `canonicalize_name`, which cannot be moved into a type-checking block, is imported, too.

### PR DESCRIPTION
**Pull request by sondrelg**
Closes #137

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -137,6 +137,25 @@ modules to ignore.
 type-checking-exempt-modules = typing_extensions  # default []
 ```
 
+### Strict
+
+The plugin, by default, will report TC001 errors
```
Comment (radoering) [inline README.md:142]: It's not only `TC001`, but also `TC002` and `TC003`, isn't it?

File: tests/test_strict.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import textwrap
+
+from flake8_type_checking.constants import TC002
+from tests.conftest import _get_error
+
+
+def test_strict_mode():
+    """
+    Assert that imports are flagged for TC001 on a per-module basis by default,
```
Comment (radoering) [inline tests/test_strict.py:9]: It seems the test is for `TC002`.

**Review by radoering — COMMENTED** [review]
It works as expected.

Apart from some minor documentation nitpicks, LGTM. 👍

**Comment by sondrelg** [comment]
Yep you're right about both. Thanks! 

I'll merge this now, and try to dig into #139 a little before releasing v2.2.0

**Pull request merged**